xbmc/cores/RetroPlayer/rendering/VideoRenderers/RPWinRenderer.cpp

```diff
@@ -143,17 +143,178 @@ bool CWinRenderBuffer::CreateScalingContext()
   return true;
 }
 
-void CWinRenderBuffer::ScalePixels(uint8_t* source,
+void CWinRenderBuffer::ScalePixels(const uint8_t* source,
                                    unsigned int sourceStride,
                                    uint8_t* target,
                                    unsigned int targetStride)
 {
-  uint8_t* src[] = {source, nullptr, nullptr, nullptr};
-  int srcStride[] = {static_cast<int>(sourceStride), 0, 0, 0};
-  uint8_t* dst[] = {target, nullptr, nullptr, nullptr};
-  int dstStride[] = {static_cast<int>(targetStride), 0, 0, 0};
 
-  sws_scale(m_swsContext, src, srcStride, 0, m_height, dst, dstStride);
+  // libav doesn't provide a pixel format for R10G10B10A2, so we have to do
+  // it ourselves.
+  if (m_targetDxFormat == DXGI_FORMAT_R10G10B10A2_UNORM)
```

Comment:
What about this? https://github.com/FFmpeg/FFmpeg/blob/50f34172e0cca2cabc5836308ec66dbf93f5f2a3/libavutil/hwcontext_d3d11va.c#L90

Reply:
I tried this initially. It gives the wrong colors/alpha:

![screenshot00009](https://github.com/xbmc/xbmc/assets/531482/0c6ae9e0-1d49-4e85-8366-d837936577de)

Reply:
Sorry, I tried `AV_PIX_FMT_X2RGB10`, as `AV_PIX_FMT_X2BGR10` isn't available in our current libav version. But I don't think it either format works.

Reply:
I tested `AV_PIX_FMT_X2BGR10` and it worked! I see how the endianness of libav and DX is reversed.

However, that still means that v20 is broken. What if we upstream `AV_PIX_FMT_X2BGR10 ` and backport the scaler I wrote?